dev-db/pgadmin4-4.21 version bump

xp-vit opened the pull request:
Signed-off-by: Viktar Patotski <xp.vit.blr@gmail.com>
Bug: https://bugs.gentoo.org/720468
Closes: https://bugs.gentoo.org/720468

thesamesam commented on dev-db/pgadmin4/pgadmin4-4.21.ebuild:
```diff
@@ -0,0 +1,120 @@
+
+LICENSE="POSTGRESQL"
+SLOT="0"
+KEYWORDS="amd64 x86"
```
Don't commit straight to stable.

thesamesam replied:
By the way, repoman would have prevented this.

xp-vit replied:
Thanks a lot for pointing it out! I will definitelly try moving to `repoman commit` functionality instead of plain `git`. Fixed now.

thesamesam commented on dev-db/pgadmin4/pgadmin4-4.21.ebuild:
```diff
@@ -0,0 +1,120 @@
+
+PATCHES=( "${FILESDIR}"/${P}-python-3.8.patch )
+
+S="${WORKDIR}"/${P}/runtime
```
S defaults to ${WORKDIR}/${P}, so you could do S=${S}/runtime.

xp-vit replied:
Done!

xp-vit commented:
@thesamesam Thanks a lot for your review. Fixed all comments and added patch for Python 3.8 installation. Also tested installation with Python 3.6, 3.7, 3.8 using ebuildtester. But pgAdmin4 is still crashes on runtume due to: https://github.com/gentoo/gentoo/pull/15618 not yet merged.

thesamesam commented:
No problem, looking good now!

titanofold commented on dev-db/pgadmin4/pgadmin4-4.21.ebuild:
```diff
@@ -0,0 +1,120 @@
+"
+RDEPEND="${COMMON_DEPEND}
+	$(python_gen_cond_dep '
+		>=app-text/htmlmin-0.1.12[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/blinker-1.4[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-1.0.2[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-gravatar-0.5.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-htmlmin-1.5.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-login-0.4.1[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-mail-0.9.1[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-migrate-2.4.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-paranoid-0.2.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-principal-0.4.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-security-3.0.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-sqlalchemy-2.3.2[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/flask-wtf-0.14.2[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/passlib-1.7.1[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/psutil-5.5.1[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/psycopg-2.8[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/python-dateutil-2.8.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/pytz-2018.9[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/simplejson-3.16.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/six-1.12.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/speaklater-1.3[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/sqlalchemy-1.2.18[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/sqlparse-0.2.4[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/sshtunnel-0.1.4[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/wtforms-2.2.1[${PYTHON_MULTI_USEDEP}]
```
Some of these need to be updated. Check the `requirements.txt` in the tarball.

xp-vit replied:
Done, but before merging this we need to bump sqlalchemy.

xp-vit commented:
It's already updated to 4.25 in upstream. No sense to keep it open.
